pkg/skaffold/deploy/deploy_mux.go

```diff
@@ -86,3 +87,42 @@ func (m DeployerMux) Render(ctx context.Context, w io.Writer, as []graph.Artifac
 	allResources := strings.Join(resources, "\n---\n")
 	return manifest.Write(strings.TrimSpace(allResources), filepath, w)
 }
+
+func (m DeployerMux) StartLogger(ctx context.Context, out io.Writer, namespaces []string) error {
+	for _, deployer := range m {
```

Comment:
Iterating all the component `deployer`s and calling their `StartLogger`, but then handling the multiple invocations with a `sync.Once` seems wrong somehow in the same PR 😅
I don't really have an idea for a better approach.

Reply:
I was looking into `ContainerManager` for #5932. It occurred to me, we could follow the pattern `ContainerManager` has. 
What if, we keep a map of providerID and instances of log providers in this `DeployMux` ?
```
type DeployerMux struct {
d []Deployer
activeLoggers map[log.Provider]log.LogProvider
}
```

And then startLoggers, will add 

```
func (m DeployerMux) StartLogger(ctx context.Context, out io.Writer, namespaces []string) error {	
  for _, deployer := range m.d {
    if id ok := deployer.LogProviderId, !ok {
       d.actiiveLoggers[id] := deployer.GetLogger()
       d.actiiveLoggers[id].StartLogger()
    }
  }
}
```

The DeployerMux now also handles making sure there is unique instance of log provider.